analysis/topEFT/topeft.py

```diff
@@ -126,16 +126,16 @@ def process(self, events):
         if isData:
             self._do_systematics = False
         else:
-            sow_ISRUp          = self._samples[dataset]["nSumOfWeights_ISRUp"]
-            sow_ISRDown        = self._samples[dataset]["nSumOfWeights_ISRDown"]
-            sow_FSRUp          = self._samples[dataset]["nSumOfWeights_FSRUp"]
-            sow_FSRDown        = self._samples[dataset]["nSumOfWeights_FSRDown"]
-            sow_renormUp       = self._samples[dataset]["nSumOfWeights_renormUp"]
-            sow_renormDown     = self._samples[dataset]["nSumOfWeights_renormDown"]
-            sow_factUp         = self._samples[dataset]["nSumOfWeights_factUp"]
-            sow_factDown       = self._samples[dataset]["nSumOfWeights_factDown"]
-            sow_renormfactUp   = self._samples[dataset]["nSumOfWeights_renormfactUp"]
-            sow_renormfactDown = self._samples[dataset]["nSumOfWeights_renormfactDown"]
+            sow_ISRUp          = self._samples[dataset]["nSumOfWeights_ISRUp"          ]if "nSumOfWeights_ISRUp"          in self._samples[dataset] else -1
+            sow_ISRDown        = self._samples[dataset]["nSumOfWeights_ISRDown"        ]if "nSumOfWeights_ISRDown"        in self._samples[dataset] else -1
+            sow_FSRUp          = self._samples[dataset]["nSumOfWeights_FSRUp"          ]if "nSumOfWeights_FSRUp"          in self._samples[dataset] else -1
+            sow_FSRDown        = self._samples[dataset]["nSumOfWeights_FSRDown"        ]if "nSumOfWeights_FSRDown"        in self._samples[dataset] else -1
+            sow_renormUp       = self._samples[dataset]["nSumOfWeights_renormUp"       ]if "nSumOfWeights_renormUp"       in self._samples[dataset] else -1
+            sow_renormDown     = self._samples[dataset]["nSumOfWeights_renormDown"     ]if "nSumOfWeights_renormDown"     in self._samples[dataset] else -1
+            sow_factUp         = self._samples[dataset]["nSumOfWeights_factUp"         ]if "nSumOfWeights_factUp"         in self._samples[dataset] else -1
+            sow_factDown       = self._samples[dataset]["nSumOfWeights_factDown"       ]if "nSumOfWeights_factDown"       in self._samples[dataset] else -1
+            sow_renormfactUp   = self._samples[dataset]["nSumOfWeights_renormfactUp"   ]if "nSumOfWeights_renormfactUp"   in self._samples[dataset] else -1
+            sow_renormfactDown = self._samples[dataset]["nSumOfWeights_renormfactDown" ]if "nSumOfWeights_renormfactDown" in self._samples[dataset] else -1
```

Comment:
Was this introduced while you were testing? Should we undo these changes, or even better, raise an exception?

Reply:
I believe all the jsons in the repo (at least for the ND T3) have the sum of weights stored. We should add them to any missing files (like the PSI version), otherwise the systematics will be wrong.

Reply:
indeed, but there are samples for which these weights are not even computed (for instance the QCD ones for the nonprompt estimation), so there's nothing to be put in the `json`. 
Of course in that case one cannot expect the systematics to work and probably we don't want to put this behavior as part of the "mainstream" code. I've rolled back the code to the previous version, but  it's reading only the weights when one wants to compute the systematics. That way i can run these samples (mostly to cross check non prompt estimations, for which I don't need these kind of systematic) and at the same time we prevent mistakes from miscalculated uncertainties